build/build-java.yml

```diff
@@ -0,0 +1,87 @@
+pr:
+  branches:
+    include:
+      - main
+  paths:
+    include:
+      - java-library/*
+      - samples/*/java/*
+    exclude:
+      - '**.md'
+
+trigger:
+  branches:
+    include:
+      - main
+  paths:
+    include:
+      - java-library/*
+      - samples/*/java/*
+    exclude:
+      - '**.md'
+
+stages:
+  - stage: BuildJava
+    displayName: 'Build Java'
+    dependsOn: []
+    variables:
+      Codeql.BuildIdentifier: java_library_samples
+      Codeql.Language: java
+    jobs:
+      - job: BuildJavaLibraryAndSamples
+        displayName: Build Java Library and Samples
+
+        pool:
+          vmImage: 'windows-latest'
+
+        steps:
+          - script: npm install -g azure-functions-core-tools
+            displayName: 'Install Azure Functions Core Tools'
+
+          - task: Maven@3
+            displayName: Build library
+            inputs:
+              mavenPomFile: java-library/pom.xml
+              options: --batch-mode --define=org.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn --update-snapshots
+              checkStyleRunAnalysis: true
+
+          - task: Maven@3
+            displayName: Build Chat Sample
```

Comment:
Can we move building sample as a separate job within this template rather than mixing it? If they depend on the library, make it as dependent on, but that will ensure we do a clean build, test and package of library (annotations) itself before we go into samples.

Reply:
I have added a separate job for samples.
But in this pipeline, we are not using the same library version, for that we will need to publish the library locally and make samples consume from local source.

Right now, samples are using maven as a source.

Reply:
added an issue to handle this separately.